Update readme for the repo

lunaceee opened the pull request:
Closes https://github.com/netlify/netlify-react-ui/issues/1866

lunaceee committed:
wip update readme for Functions code examples

lunaceee committed:
added illo to readme

lunaceee committed:
added readme file for functions examples

lunaceee committed:
wip

lunaceee committed:
Merge branch 'master' into update-readme-all

verythorough commented on README.md:
```diff
@@ -1,19 +1,3 @@
-## use-env
-
-This is a basic example of a function that uses a build environment variable that's specified in the users Netlify site's deploy settings. [Learn more](https://github.com/netlify/code-examples/tree/update-readme/function_examples/use-env).
+Code snippets for you to get started with Netlify features.
```
I think this page should include a little more guidance for the reader - links to each folder, or at least an explanation that each folder contains another example with a readme of its own. (Right now the list would have only one item - the function_examples folder, but that's OK.)

depadiernos committed:
Clarified Step 3 in the Localhost section

Added that to test function on the website side, the user should go to localhost:1234 and type an address and they should check in the browser dev tools

lunaceee committed:
Merge pull request #7 from netlify/token-hider-readme

Clarified Step 3 in the Localhost section

verythorough commented on function_examples/README.md:
```diff
@@ -0,0 +1,25 @@
+
+## Basic code examples
+
+To help you quickly get started, we created several simple code examples under `functions_examples` folder to showcase the basic usage of Netlify Functions. Detailed instructions enclosed under each subfolder.
```
This readme is in the `function_examples` folder already, so I don't think this explanation is necessary. Instead, I would make a list of links to each subfolder, with the name and a very short description of the sample's key programming characteristics. For example:

- [Random Word](/random-word): a very basic function with no dependencies or build tools
- [Use Env](/use-env): function that uses Netlify build environment variables, with no dependencies or build tools
- [Rest](/rest): function for calling an external API with `node-fetch`, using `netlify-lambda` to bundle dependencies
- [Token Hider](/token-hider): function using `axios` to access an external API, using Netlify build environment variables to hide the API token; bundled with `netlify-lambda`

verythorough commented on function_examples/random-word/README.md:
```diff
@@ -1,3 +1,3 @@
-Basic Examples of using netlify functions
+# Random word
 
+This is a basic example of a function that when it gets called, returns a random word from an array of words back. This function has no build process and does not use any external packages.
```
I would add that the function itself is stored in `/functions`, and the page with a link that calls the function is in `/_site`. I assume @futuregerald also made a demo site with this page, so I would link that so people can see how it works.

verythorough commented on function_examples/rest/README.md:
```diff
@@ -1,2 +1,7 @@
+
+It includes an `index.html` that makes the call and displays the response.
+
+This function gets data from https://jsonplaceholder.typicode.com/, and also has a build process.
```
I would say that netlify-lambda is what's being used for the build process, and that it takes the source files and package.json from `/func` and builds the function as a single file in a new folder, `/functions`. I would link to netlify-lambda for further information on local serving, build commands, etc.

verythorough commented on function_examples/token-hider/README.md:
```diff
@@ -17,24 +17,27 @@ When you deploy your site to Netlify, you'll need to define the following [build
-
-Type in the following script in your console:
 `API_URL="https://maps.googleapis.com/maps/api/geocode/json" API_TOKEN="your Google Maps API key" npm run lambda:dev`
+
```
Something is wrong with this line. I think some explanatory text may have been accidentally removed at some point. I'd check with @depadiernos to see what his intention was here.

verythorough replied:
(I'm guessing that at some point there was a step to add those two variables to a `.env` file before running the `npm run lambda:dev` line at the end of the line.)

verythorough replied:
I didn't realize this command would work! I think it's worth telling the reader to run these lines in the command line.

verythorough commented on function_examples/token-hider/README.md:
```diff
@@ -17,24 +17,27 @@ When you deploy your site to Netlify, you'll need to define the following [build
 The site is now available at `http://localhost:1234/`
 
-Open the site in your browser and verify that the API key is not anywhere on the client side.
+_Notes: Make sure you run Functions first before you run the site._
```
This is included in step 2 (line 23), but something weird with formatting combined the command with some variables. When Step 2 is fixed, I don't think this line will be necessary.

verythorough commented on function_examples/use-env/README.md:
```diff
@@ -1,5 +1,5 @@
-_at this moment you can only use variables specified in the UI and not in netlify.toml_
+This is a basic example of a function that uses a build environment variable that's specified in the users Netlify site's deploy settings.
 
+_at this moment you can only use variables specified in the UI and not in netlify.toml_
```
Capitalize the `a` at the beginning and add a period at the end.

Also, this may no longer be true. @calavera could probably confirm.

verythorough reviewed (changes requested):
I'm glad you're looking at these and testing them out! I made a few suggestions for improving the content.

lunaceee committed:
revised readme for the code examples repo

lunaceee committed:
revised readme for functions folder

lunaceee committed:
updated readme for token hider

lunaceee committed:
updated readme for random-word, rest and use-env

lunaceee committed:
fixed conflict

lunaceee committed:
wip polished copies

lunaceee committed:
removed illo and added demo site to random word example

lunaceee commented:
I've updated the files based on your review @verythorough . Let me know if it looks good to you. Thanks!
